src/charm.py

```diff
@@ -179,10 +180,20 @@ def _create_memory_storage(self, path: Path, size: int) -> None:
             else:
                 execute_command(["mount", "-o", f"remount,size={size}k", str(path)])
         except OSError as err:
-            logger.exception("Unable to create directory")
+            logger.exception("Unable to create storage directory")
+            # Ensure the path is empty for next retry.
+            if path.exists():
+                shutil.rmtree(path, ignore_errors=True)
+                path.rmdir()
+                logger.info("Cleaned up storage directory")
```

Comment:
I wonder if a bit of factoring could help here, it would avoid repeating but also ensure that any change will be applied to the 2 parts where this is required